finetuner/tuner/callback/best_model_checkpoint.py

```diff
@@ -0,0 +1,128 @@
+import os
+from typing import TYPE_CHECKING
+
+import keras
+import numpy as np
+import paddle
+import torch
+from jina.logging.logger import JinaLogger
+
+from finetuner.helper import get_framework
+
+from .base import BaseCallback
+
+if TYPE_CHECKING:
+    from ..base import BaseTuner
+
+
+class BestModelCheckpoint(BaseCallback):
+    """
+    Callback to save the best model across all epochs
+
+    An option this callback provides include:
+    - Definition of 'best'; which quantity to monitor and whether it should be
+        maximized or minimized.
+    """
+
+    def __init__(
+        self,
+        save_dir: str,
+        monitor: str = 'val_loss',
+        mode: str = 'auto',
+    ):
+        """
+        :param save_dir: string, path to save the model file.
+        :param monitor: if `monitor='loss'` best bodel saved will be according
+            to the training loss, if `monitor='val_loss'` best model saved will be
+            according to the validation loss
+        :param mode: one of {'auto', 'min', 'max'}. the
+            decision to overwrite the current save file is made based on either
+            the maximization or the minimization of the monitored quantity.
+            For `val_acc`, this should be `max`, for `val_loss` this should be
+            `min`, etc. In `auto` mode, the mode is set to `max` if the quantities
+            monitored are 'acc' or start with 'fmeasure' and are set to `min` for
+            the rest of the quantities.
+        """
+        self._logger = JinaLogger(self.__class__.__name__)
+        self._save_dir = save_dir
+        self._monitor = monitor
+        self._train_losses = []
+        self._valid_losses = []
+        if not save_dir:
+            raise ValueError(
+                '``save_dir`` parameter is mandatory. Pass it in parameters'
+            )
```

Comment:
I think you can delete this. You already made `save_dir` a required parameter by nor providing default